File: app/Models/Beneficiary.php
<?php

namespace App\Models;

use App\Traits\SyncsWithHQ;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Beneficiary extends Model
{
    use HasFactory, SoftDeletes, SyncsWithHQ;

    protected $fillable = [
        'user_id', 'name', 'nickname', 'bank_name', 'account_number',
        'routing_number', 'swift_code', 'iban', 'type', 'currency',
        'is_favorite', 'is_verified',
    ];

    protected function casts(): array
    {
        return [
            'is_favorite' => 'boolean',
            'is_verified' => 'boolean',
        ];
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function scheduledPayments()
    {
        return $this->hasMany(ScheduledPayment::class);
    }

    public function scopeFavorites($query)
    {
        return $query->where('is_favorite', true);
    }

    public function getDisplayNameAttribute(): string
    {
        return $this->nickname ?: $this->name;
    }

    public function getMaskedAccountAttribute(): string
    {
        return '****' . substr($this->account_number, -4);
    }
}

File: mobile/lib/data/models/beneficiary.dart
class Beneficiary {
  final int id;
  final String name;
  final String? nickname;
  final String? bankName;
  final String accountNumber;
  final String type;
  final String currency;
  final bool isFavorite;
  final bool isVerified;

  const Beneficiary({
    required this.id,
    required this.name,
    this.nickname,
    this.bankName,
    required this.accountNumber,
    required this.type,
    required this.currency,
    this.isFavorite = false,
    this.isVerified = false,
  });

  factory Beneficiary.fromJson(Map<String, dynamic> json) {
    return Beneficiary(
      id: json['id'] as int,
      name: json['name'] as String,
      nickname: json['nickname'] as String?,
      bankName: json['bank_name'] as String?,
      accountNumber: json['account_number'] as String,
      type: json['type'] as String,
      currency: json['currency'] as String,
      isFavorite: json['is_favorite'] as bool? ?? false,
      isVerified: json['is_verified'] as bool? ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'nickname': nickname,
      'bank_name': bankName,
      'account_number': accountNumber,
      'type': type,
      'currency': currency,
      'is_favorite': isFavorite,
      'is_verified': isVerified,
    };
  }

  String get displayName => nickname ?? name;
}
